src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
So right now you can accomplish that with a dedicated custom ASOS middleware that always calls into that schemes process request, or you can override the IAuthenticationSchemeProvider's GetRequestHandlerScheme method to always return the ASOS scheme for all requests.